packages/mdc-textfield/foundation.js

```diff
@@ -247,8 +280,11 @@ class MDCTextfieldFoundation extends MDCFoundation {
     if (isValid) {
       this.adapter_.removeClass(INVALID);
     } else {
-      this.adapter_.addClassToLabel(LABEL_SHAKE);
       this.adapter_.addClass(INVALID);
+
+      if (this.isFocused_) {
```

Comment:
is there a test for this case? A test for when the text fields value is changed programmatically, but the user is not focused on that text field. You should verify that `this.adapter_.addClassToLabel(LABEL_SHAKE)` is not called.